ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -70,6 +70,34 @@ type domain_info = {
         (* Persist netbios_name to support hostname change *)
 }
 
+module KerberosEncryptionTypes = struct
```

Comment:
i believe the purpose of this module is to validate `Xapi_globs.winbind_kerberos_encryption_type`, which you do in just one function:

```
let get_encryption_type () =
  let s = !Xapi_globs.winbind_kerberos_encryption_type in
  if List.mem s ["strong"; "all"; "legacy" ] then s else ( debug "some warning" ;  "strong" )
```

i'm not sure i see the benefit of the variant type